### DISCUSSION TITLE
Reduce MQTT traffic (Tuya module)

### DISCUSSION THREAD

**Discussion by benjaminnestler** [arendst/Tasmota#15456] [Q&A] [answered]
Hi

we want to reduce the mqtt traffic, because mqtt messages in AWS cloud instances leads to costs. We successfully connect some Tuya devices so we concentrate on the module **xdrv_16_tuyamcu.ino**. Every (TuyaSend)action we perform via mqtt will be result in a response wich is created by function `ResponseCmndDone()` (**_"TuyaSend Done"_**). This response isn't necessary and causes costs. 

Every action event on a Tuya device, leads to a reported messages from tuya host controller. The reported message will be received by tasmota and could be send via mqtt (SetOption66 = 1). 

Whats your opinion? 

Could I add a new SetOption-feature to suppress the `ResponseCmndDone()` function in **xdrv_16_tuyamcu.ino**?

Maybe this could be a feature which could be implemented in other modules as well?

**Comment by sfromis** [comment, marked as answer]
Suppressing result messages breaks rules and can impact other automation options. In general the "cost" of a few messages you do not have a use case for is really negligible.

**Comment by benjaminnestler** [comment]
Hi sfromis,

thanks four your answer. Costs of a few messages is negligible for one device thats right. But think about thousands of devices which are connected with a cloud. Every single message which will be send by these thousand devices creates mqtt traffic. 

See the following example:

Imagine you are the operator of cloud services and 10.000 devices are connected to your cloud. Every device response only one message (triggered by `ResponseCmndDone()` function) per hour per day. Monthly this creates a message traffic 1 x 24 x 30 x 10.000 = 7.200.000 messages.

So every single message counts, reducing the mqtt traffic ...

**Reply by barbudor** [reply]
So aws cost is $1 per millions of messages so your overall cost for 10'000 is less than $8

If you have a use case of 10'000 devices you have the business that goes with it and $8 shouldn't matter much
